app/src/full/kotlin/io/homeassistant/companion/android/launch/LaunchPresenterImpl.kt

```diff
@@ -5,34 +5,36 @@ import dagger.hilt.android.qualifiers.ActivityContext
 import dagger.hilt.android.scopes.ActivityScoped
 import io.homeassistant.companion.android.BuildConfig
 import io.homeassistant.companion.android.common.data.integration.DeviceRegistration
+import io.homeassistant.companion.android.common.data.network.NetworkStatusMonitor
 import io.homeassistant.companion.android.common.data.servers.ServerManager
 import io.homeassistant.companion.android.onboarding.getMessagingToken
 import javax.inject.Inject
-import kotlinx.coroutines.launch
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
 import timber.log.Timber
 
 @ActivityScoped
-class LaunchPresenterImpl @Inject constructor(@ActivityContext context: Context, serverManager: ServerManager) :
-    LaunchPresenterBase(context as LaunchView, serverManager) {
-    override fun resyncRegistration() {
-        ioScope.launch {
-            if (!serverManager.isRegistered()) return@launch
-            serverManager.defaultServers.forEach {
-                ioScope.launch {
-                    try {
-                        serverManager.integrationRepository(it.id).updateRegistration(
-                            DeviceRegistration(
-                                "${BuildConfig.VERSION_NAME} (${BuildConfig.VERSION_CODE})",
-                                null,
-                                getMessagingToken(),
-                            ),
-                        )
-                        serverManager.integrationRepository(it.id).getConfig() // Update cached data
-                        serverManager.webSocketRepository(it.id).getCurrentUser() // Update cached data
-                    } catch (e: Exception) {
-                        Timber.e(e, "Issue updating Registration")
-                    }
-                }
+class LaunchPresenterImpl @Inject constructor(
+    @ActivityContext context: Context,
+    serverManager: ServerManager,
+    networkStatusMonitor: NetworkStatusMonitor,
+) : LaunchPresenterBase(context as LaunchView, serverManager, networkStatusMonitor) {
+    override suspend fun resyncRegistration() = withContext(Dispatchers.IO) {
```

Comment:
Actually I don't think we need to change the dispatcher here, it was using IO before because we didn't have any scope, but now with this change we can delegate to whatever is calling. The network and DB call are made on a dedicated dispatcher already.